quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -118,18 +119,44 @@ private static boolean requiresReAugmentation(CommandLine cmd) {
         if (hasConfigChanges()) {
             cmd.getOut().println("Changes detected in configuration. Updating the server image.");
             if(!isDevMode()) {
+                List<String> cliInput = getSanitizedCliInput();
                 cmd.getOut().printf("For an optional runtime and bypass this step, please run the '%s' command prior to starting the server:%n%n\t%s %s %s%n",
                         Build.NAME,
                         Environment.getCommand(),
                         Build.NAME,
-                        String.join(" ", asList(ARG_SPLIT.split(Environment.getConfigArgs()))) + "\n");
+                        String.join(" ", cliInput) + "\n");
             }
             return true;
         }
 
         return hasProviderChanges();
     }
 
+    /**
+     * checks the raw cli input for possible credentials / properties which should be masked,
+     * and masks them.
+     * @return a list of potentially masked properties in CLI format, e.g. `--db-password=*******`
+     * instead of the actual passwords value.
+     */
+    private static List<String> getSanitizedCliInput() {
+
+        if(Environment.getConfigArgs().isEmpty()) {
+            return List.of(Environment.getConfigArgs());
+        }
+
+        List<String> rawCliArgs = asList(ARG_SPLIT.split(Environment.getConfigArgs()));
+        List<String> properties = new ArrayList<>();
+
+        if(!rawCliArgs.isEmpty()) {
+            for(String rawCliArg : rawCliArgs) {
+                String keyInCliFormat = rawCliArg.split("=")[0];
+                String propertyKey = MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX + keyInCliFormat.replace(ARG_PREFIX, "").replace("-", ".");
```

Comment:
What if you have an option like `hostname-strict-https`? Looks like it will formatted as `hostname.strict.https`? But the property format should be `hostname.strict-https`

Reply:
ohhh you are right. will provide a fix.